tests/integration/test_bundle.py

```diff
@@ -276,6 +278,35 @@ async def test_loki_receives_logs(ops_test: OpsTest):
     logger.info("loki is successfully receiving flog logs")
 
 
+@pytest.mark.abort_on_fail
+async def test_all_traefik_proxied_urls_return_200(ops_test: OpsTest):
+    """Verify that all ingressed applications' URLs are reachable from localhost and return 200."""
+    proxied_urls = await get_all_proxied_urls(ops_test)
+    for app, url in proxied_urls.items():
+        response = urlopen(url, context=insecure_context)
```

Comment:
Turns out [urlopen](https://docs.python.org/3.8/library/urllib.request.html#urllib.request.urlopen) raises for >=400.
In a [test failure](https://github.com/canonical/cos-lite-bundle/actions/runs/8724158615/job/23934108235?pr=93#step:7:663) we get 
```
  File "/home/runner/work/cos-lite-bundle/cos-lite-bundle/tests/integration/test_bundle.py", line 286, in test_all_traefik_proxied_urls_return_200
...
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```
Might be a good idea to introduce try-except (with a bare exception) so we see in the error log which URL failed with a 404. And/or add a `logger.info("Querying %s over %s", app, url)` before urlopen.